### DISCUSSION TITLE
Migration strategy of DMS V1 to V2 [Terraform]

### DISCUSSION THREAD

**Discussion by bboron86** [opentelekomcloud/terraform-provider-opentelekomcloud#2586] [Q&A]
Hi,

we're running a Kafka v2.7 with Terraform's `opentelekomcloud_dms_product_v1` provider. However the `V1` API is now deprecated and we should migrate to `V2`. How can we migrate `terraform state` to the new API,  WITHOUT recreating all kafka resources?

**Comment by anton-sidelnikov** [comment]
new DMS resource is introduced opentelekomcloud_dms_dedicated_instance_v2.
It doesn't have any deprecated API and you can import it by using `terraform import` and id of your DMS instance with a couple of lifecycle `ignore_changes`(look at documentation).
